### PR TITLE
Completing glossary translation for Bahasa pt. 7

### PR DESCRIPTION
**Pull request by habibrosyad**
Part of #24382 

Updated:
- image
- kube-apiserver

New:
- extensions
- init-container
- job

/cc @ariscahyadi @girikuncoro

### CODE REVIEW

File: content/id/docs/reference/glossary/extensions.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+aka:
+tags:
+- fundamental
+- extension
+---
+Ekstensi adalah komponen perangkat lunak yang memperluas dan terintegrasi secara mendalam dengan Kubernetes untuk mendukung perangkat keras baru.
+
+<!--more-->
+
+Sebagian besar admin klaster akan menggunakan instans Kubernetes yang dihos atau didistribusikan. Akibatnya, hampir semua pengguna Kubernetes perlu menginstal [ekstensi](/id/docs/concepts/extend-kubernetes/extend-cluster/#perluasan) dan sedikit pengguna yang perlu membuat ekstensi baru.
```
Comment (ariscahyadi) [inline content/id/docs/reference/glossary/extensions.md:17]: Same with PR #24427. We need to keep “instance”.
nit! “didistribusikan” is not good translation for “distribution”.
```suggestion
Sebagian besar admin klaster akan menggunakan _instance_ Kubernetes yang dihoskan (_hosted_) atau didistribusikan. Akibatnya, hampir semua pengguna Kubernetes perlu menginstal [ekstensi](/id/docs/concepts/extend-kubernetes/extend-cluster/#perluasan) dan sedikit pengguna yang perlu membuat ekstensi baru.
```

File: content/id/docs/reference/glossary/image.md [hunk 2]
```diff
@@ -4,15 +4,14 @@ id: image
 date: 2019-04-24
 full_link: 
 short_description: >
-  Instans yang disimpan dari sebuah kontainer yang mengandung seperangkat perangkat lunak yang dibutuhkan untuk menjalankan sebuah aplikasi.
+  Instans yang disimpan dari sebuah Container yang memuat seperangkat perangkat lunak yang dibutuhkan untuk menjalankan sebuah aplikasi.
```
Comment (ariscahyadi) [inline content/id/docs/reference/glossary/image.md:7]: Same with PR #24427. We need to keep “instance“.

File: content/id/docs/reference/glossary/image.md [hunk 3]
```diff
@@ -4,15 +4,14 @@ id: image
...
 
-aka: 
+aka:
 tags:
 - fundamental
 ---
- Instans yang disimpan dari sebuah kontainer yang mengandung seperangkat perangkat lunak yang dibutuhkan untuk menjalankan sebuah aplikasi.
+Instans yang disimpan dari sebuah Container yang memuat seperangkat perangkat lunak yang dibutuhkan untuk menjalankan sebuah aplikasi.
```
Comment (ariscahyadi) [inline content/id/docs/reference/glossary/image.md:13]: Same with PR #24427. We need to keep “instance“.

File: content/id/docs/reference/glossary/image.md [hunk 4]
```diff
@@ -4,15 +4,14 @@ id: image
...
 
-<!--more--> 
-
-Sebuah mekanisme untuk mengemas perangkat lunak yang mengizinkan perangkat lunak tersebut untuk disimpan di dalam registri kontainer, di-_pull_ kedalam filesystem lokal, dan dijalankan sebagai suatu aplikasi. Meta data yang dimasukkan mengindikasikan _executable_ apa sajakah yang perlu dijalanmkan, siapa yang membuat _executable_ tersebut, dan informasi lainnya.
+<!--more-->
 
+Sebuah mekanisme untuk mengemas perangkat lunak yang memungkinkan perangkat lunak tersebut untuk disimpan di dalam register Container, ditarik kedalam sistem lokal, dan dijalankan sebagai suatu aplikasi. Metadata disertakan di dalam _image_ yang mengindikasikan _executable_ apa saja yang perlu dijalankan, siapa yang membuatnya, dan informasi lainnya.
```
Comment (ariscahyadi) [inline content/id/docs/reference/glossary/image.md:17]: nit!
```suggestion
Sebuah mekanisme untuk mengemas perangkat lunak yang memungkinkan perangkat lunak tersebut untuk disimpan di dalam register Container, ditarik ke dalam sistem lokal, dan dijalankan sebagai suatu aplikasi. Metadata disertakan di dalam _image_ yang mengindikasikan _executable_ apa saja yang perlu dijalankan, siapa yang membuatnya, dan informasi lainnya.
```

File: content/id/docs/reference/glossary/kube-apiserver.md [hunk 5]
```diff
@@ -4,16 +4,15 @@ id: kube-apiserver
...
 - fundamental
 ---
- Komponen di master yang mengekspos API Kubernetes. Merupakan <i> front-end </i> dari <i> kontrol plane </i> Kubernetes.
+Komponen _control plane_ yang mengekspos API Kubernetes. Merupakan _front-end_ dari _control plane_ Kubernetes.
 
 <!--more-->
 
-Komponen ini didesain agar dapat di-<i>scale</i> secara horizontal. Lihat [Membangun Klaster HA](/docs/admin/high-availability/).
-
+Komponen ini didesain agar dapat diskalakan secara horizontal. Lihat [Membangun Klaster HA](/docs/admin/high-availability/).
```
Comment (ariscahyadi) [inline content/id/docs/reference/glossary/kube-apiserver.md:18]: “horizontal” is not standard.
ID link is available.
```suggestion
Komponen ini didesain agar dapat diskalakan secara horisontal. Lihat [Membangun Klaster HA](/id/docs/admin/high-availability/).
```

**Review by ariscahyadi — COMMENTED** [review]
Thanks @habibrosyad, please address the comments.

File: content/id/docs/reference/glossary/extensions.md (hunk 1 above)
Comment (habibrosyad) [reply content/id/docs/reference/glossary/extensions.md:17]: hmm, it's based on RSNI https://github.com/jk8s/sig-docs-id/blob/master/resources/RSNI-glossarium.pdf, don't we refer to this doc too for translation?

File: content/id/docs/reference/glossary/kube-apiserver.md (hunk 5 above)
Comment (habibrosyad) [reply content/id/docs/reference/glossary/kube-apiserver.md:18]: It's the other way around... horisontal is the non official word, as described here https://kbbi.kemdikbud.go.id/entri/horizontal

**Comment by habibrosyad** [comment]
Summary notes for the review comments:
- `instans` is defined here https://github.com/jk8s/sig-docs-id/blob/master/resources/RSNI-glossarium.pdf as translation for `instance`, and it's pretty much of what `instance` means in English IT term. Or we don't actually refer to RSNI anymore? @girikuncoro 
- `horizontal` is the standard word, refer to https://kbbi.kemdikbud.go.id/entri/horizontal

wdyt @ariscahyadi?

**Comment by ariscahyadi** [comment]
> Summary notes for the review comments:
> 
> * `instans` is defined here https://github.com/jk8s/sig-docs-id/blob/master/resources/RSNI-glossarium.pdf as translation for `instance`, and it's pretty much of what `instance` means in English IT term. Or we don't actually refer to RSNI anymore? @girikuncoro
> * `horizontal` is the standard word, refer to https://kbbi.kemdikbud.go.id/entri/horizontal
> 
> wdyt @ariscahyadi?

Ok for "instans".
For the "horizontal", it is written as "bentuk tidak baku: horisontal".

**Comment by habibrosyad** [comment]
> For the "horizontal", it is written as "bentuk tidak baku: horisontal".

Yes that's exactly what I mean, _bentuk tidak baku_ of `horizontal` is `horisontal`. So `horizontal` is _bentuk baku_.

The same with https://kbbi.kemdikbud.go.id/entri/apotek, you would get "bentuk tidak baku: apotik". Another one https://kbbi.kemdikbud.go.id/entri/praktik for "bentuk tidak baku: praktek".

Also supported by this tweet from Ivan Lanin https://twitter.com/ivanlanin/status/853624314284789760?lang=en:
> Bentuk yang baku:
> - apotek, bukan apotik, dr bhs Belanda apotheek
> - praktik, bukan praktek, dr bhs Belanda praktijk

In KBBI, it seems the label "bentuk tidak baku" shows _bentuk tidak baku_ of the word being shown (not the other way around).

**Comment by ariscahyadi** [comment]
> > For the "horizontal", it is written as "bentuk tidak baku: horisontal".
> 
> @ariscahyadi Yes that's exactly what I mean, _bentuk tidak baku_ of `horizontal` is `horisontal`. So `horizontal` is _bentuk baku_.
> 
> The same with https://kbbi.kemdikbud.go.id/entri/apotek, you would get "bentuk tidak baku: apotik". Another one https://kbbi.kemdikbud.go.id/entri/praktik for "bentuk tidak baku: praktek".
> 
> Also supported by this tweet from Ivan Lanin https://twitter.com/ivanlanin/status/853624314284789760?lang=en:
> 
> > Bentuk yang baku:
> > 
> > * apotek, bukan apotik, dr bhs Belanda apotheek
> > * praktik, bukan praktek, dr bhs Belanda praktijk
> 
> In KBBI, it seems the label "bentuk tidak baku" shows _bentuk tidak baku_ of the word being shown (not the other way around).

Sorry @habibrosyad I thought it the other way around :)
Praktik is "bentuk tidak baku" from "praktek". 

Anyway, if you think that is the right.. I am OK.

**Comment by habibrosyad** [comment]
I have updated with some of the suggestions above, if you could check again @ariscahyadi , thanks :)

**Comment by irvifa** [comment]
Okay I think it’s good. Thanks!
/lgtm
/approve

**Review by sftim — COMMENTED** [review]
I'm surprised to see Container written with a capital C. In the English docs we usually write it all-lowercase.

File: content/id/docs/reference/glossary/image.md (hunk 2 above)
Comment (irvifa) [reply content/id/docs/reference/glossary/image.md:7]: @ariscahyadi do we use all capitalization for each term as Container, Pod, etc? Just to make sure

**Comment by habibrosyad** [comment]
> I'm surprised to see Container written with a capital C. In the English docs we usually write it all-lowercase.

@sftim during my first time joined the Indonesian localisation, I got the impression of that's the consensus of the team. As you can see, e.g. in https://github.com/kubernetes/website/pull/23987 I used to be translating the word as what you've suggested (`container` -> `kontainer`).

Any thought on this guys @irvifa @girikuncoro @ariscahyadi ?

**Comment by ariscahyadi** [comment]
> > I'm surprised to see Container written with a capital C. In the English docs we usually write it all-lowercase.
> 
> @sftim during my first time joined the Indonesian localisation, I got the impression that's the consensus of the team. As you can see, e.g. in #23987 I used to be translating the word as what you've suggested (`container` -> `kontainer`).
> 
> @irvifa @girikuncoro @ariscahyadi Any thought on this?

@sftim and @habibrosyad I also find similar problem (`container` --> `kontainer`) when I made my first contribution.
Then, I just follow the consensus "all API name should be in capital letter" which is same goes to `Pod`, `Node`, and so on.
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/

Even though, most of the `container` word is written in small letter in English doc, but in this documentation almost of the `Container` words are written in capital letter :) https://kubernetes.io/docs/concepts/containers/container-environment/

**Comment by sftim** [comment]
Bear in mind that most of the time, in English, “container” _doesn't_ refer to the Go struct for defining containers in Pods, nor to the API representation of that struct. It refers to containers, the kind you can also run using Docker.

So I'd write “the runtime fetches the container image” and not “the runtime fetches the Container image”.

**Comment by habibrosyad** [comment]
In this case should we just follow the upstream English for consistency? when it written as `container` we could use `_container_`, `Container` -> `Container`, or `pod` -> `_pod_`?

wdyt @ariscahyadi @irvifa @girikuncoro?

**Comment by ariscahyadi** [comment]
> Bear in mind that most of the time, in English, “container” _doesn't_ refer to the Go struct for defining containers in Pods, nor to the API representation of that struct. It refers to containers, the kind you can also run using Docker.
> 
> So I'd write “the runtime fetches the container image” and not “the runtime fetches the Container image”.

Yes, I agree with you, it is depend on the context. Similarly with `service`, sometime we can translate into `layanan` but we need to keep as `Service`. But, I think it is little bit hard for some contributors, including me maybe :)

**Comment by ariscahyadi** [comment]
> In this case should we just follow the upstream English for consistency? when it's written as `container` we could use `_container_`, `Container` -> `Container`, or `pod` -> `_pod_`?
> 
> wdyt @ariscahyadi @irvifa @girikuncoro?

@habibrosyad  in this case, the translation page will have more italics word :)
@girikuncoro and @irvifa should decide :)

**Comment by girikuncoro** [comment]
@ariscahyadi @habibrosyad this always confuses everyone actually, since context is also subjective to readers sometimes. Before we move forward, let's try to list down the confusing words first? afaik:
* Container vs kontainer
* Service vs layanan
* Node vs _node_

what else? the rest of API objects are pretty clear to me: Pod will always be Pod, etc.

**Comment by habibrosyad** [comment]
> @ariscahyadi @habibrosyad this always confuses everyone actually, since context is also subjective to readers sometimes. Before we move forward, let's try to list down the confusing words first? afaik:
> 
> * Container vs kontainer
> * Service vs layanan
> * Node vs _node_
> 
> what else? the rest of API objects are pretty clear to me: Pod will always be Pod, etc.

@girikuncoro For this particular page https://kubernetes.io/docs/reference/access-authn-authz/rbac/:
> A role binding grants the permissions defined in a role to a user or set of users. It holds a list of subjects (users, groups, or service accounts), and a reference to the role being granted. A RoleBinding grants permissions within a specific namespace whereas a ClusterRoleBinding grants that access cluster-wide.

As far as I know, `role binding` and `RoleBinding` in this context are basically interchangeable. In my translation, instead of using `_role binding_` (it is weird to translate it into `pengikatan rol`), I used `RoleBinding` (for all occurrences). This is one of the challenges with localisation in my experience.

**Comment by ariscahyadi** [comment]
Based on my experience, especially the single word:

- Container vs kontainer
- Service vs layanan
- Node vs _node_
- Job vs pekerjaan
- Event vs kejadian or _event_
- Deployment vs Penggelaran
- Volume vs _volume_

**Pull request merged**